This patch was developed in-house. It can be eventually offered to upstream.

--- git-2.36.0/compat/inet_ntop.c
+++ git-2.36.0/compat/inet_ntop.c
@@ -168,7 +168,7 @@ * author:
  *	Paul Vixie, 1996.
  */
 const char *
-inet_ntop(int af, const void *src, char *dst, size_t size)
+inet_ntop(int af, const void *src, char *dst, socklen_t size)
 {
 	switch (af) {
 	case AF_INET:
--- git-2.36.0/git-compat-util.h
+++ git-2.36.0/git-compat-util.h
@@ -882,7 +882,7 @@ int inet_pton(int af, const char *src, v
 #endif
 
 #ifdef NO_INET_NTOP
-const char *inet_ntop(int af, const void *src, char *dst, size_t size);
+const char *inet_ntop(int af, const void *src, char *dst, socklen_t size);
 #endif
 
 #ifdef NO_PTHREADS
